build/beatAML/GetBeatAML.py

```diff
@@ -115,24 +83,80 @@ def modify_patient_file(file_path):
+    """
+    Generate samples file by reading, processing and merging multiple source files.
+
+    This function reads two Excel files, processes the contained data,
+    and generates a combined sample file in CSV format.
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing the combined samples data.
+    """
+
+    # Read the beataml_waves1to4_sample_mapping.xlsx file
+    beataml_samples = pd.read_excel('beataml_waves1to4_sample_mapping.xlsx')
+    additional_samples = beataml_samples[~beataml_samples['rna_control'].isin(['No', '']) & pd.notna(beataml_samples['rna_control'])]
+    
+    # Map rna_control to common_name and retain columns LabId, dbgap_rnaseq_sample
+    additional_samples = additional_samples[['labId', 'dbgap_rnaseq_sample', 'rna_control']]
+
+    # Read the other file
+    mapped_info = pd.read_excel('1-s2.0-S1535610822003129-mmc2.xlsx', skiprows=1)
+
+    # Merge with the original samples on dbgap_rnaseq_sample
+    original_samples = mapped_info.merge(beataml_samples, on="dbgap_rnaseq_sample", how="inner")
+
+    # Concatenate the original samples and the additional samples
+    full_samples = pd.concat([original_samples, additional_samples], ignore_index=True)
+
+    full_samples = full_samples[["labId", "specificDxAtInclusion", "specimenType","rna_control"]]
+    full_samples['common_name'] = np.where(full_samples['specimenType'].isna(), full_samples['rna_control'], full_samples['specimenType'])
+    full_samples.drop(columns=["specimenType","rna_control"], inplace=True)
+    full_samples.rename(columns={"labId": "other_id"}, inplace=True)
+    full_samples.rename(columns={"specificDxAtInclusion": "other_names"}, inplace=True)
+    full_samples['other_names'] = full_samples['other_names'].fillna('Control')
+    full_samples["cancer_type"] = "ACUTE MYELOID LEUKAEMIA"
+    full_samples["model_type"] = "ex vivo"
+    full_samples["other_id_source"] = "beatAML"
+    full_samples.drop_duplicates(subset='other_id', keep='first', inplace=True)
+    
+    prot_sample_file = 'PNNL_clinical_summary_12_08_2021_updated_OS_4patients_02_28_2022.txt'
+    modify_patient_file(prot_sample_file)
+    prot_samples = pd.read_csv(prot_sample_file, delimiter=' ', skipinitialspace=True, encoding='ISO-8859-1', quotechar='"')
+    prot_samples = prot_samples[["labId","specificDxAtInclusion","specimenType"]]
+    prot_samples.rename(columns={"labId": "other_id"}, inplace=True)
+    prot_samples.rename(columns={"specificDxAtInclusion": "other_names"}, inplace=True)
+    prot_samples.rename(columns={"specimenType": "common_name"}, inplace=True)
+    prot_samples["cancer_type"] = "ACUTE MYELOID LEUKAEMIA"
+    prot_samples["model_type"] = "ex vivo"
+    prot_samples["other_id_source"] = "beatAML"    
+    
+    all_samples = pd.concat([prot_samples, full_samples])
+    maxval = max(pd.read_csv('https://raw.githubusercontent.com/PNNL-CompBio/candleDataProcessing/main/build/hcmi/samples.csv').improve_sample_id)
```

Comment:
remove this, please only work locally, as the files should all build in the same place for now, and not require uploading to github.

Reply:
Just that last line right?  Should I change all instances where a file is downloaded from GitHub? (I’ll have it pull from the CODR repo instead)

Reply:
Sorry, I mis-remembered something on this. This is the only instance where it pulls from our Github. And I'll change it to CODR.

Reply:
yeah, i fixed up all the other scripts, this is the only one left.